### PR TITLE
LPD-15348 Escape publication name in notifications

### PR DESCRIPTION
**Pull request by ctang3**
https://liferay.atlassian.net/browse/LPD-15348

### CODE REVIEW

File: modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/notifications/ScheduledPublicationUserNotificationHandler.java [hunk 1]
```diff
@@ -61,7 +62,10 @@ protected String getBody(
 				_language.format(
 					serviceContext.getLocale(),
 					"x-scheduled-publication-failed",
-					new Object[] {jsonObject.getString("ctCollectionName")},
+					new Object[] {
+						HtmlUtil.escape(
+							jsonObject.getString("ctCollectionName"))
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/notifications/ScheduledPublicationUserNotificationHandler.java:67]: Do we need to escape this twice? You're already escaping it in `CTPublishBackgroundTaskExecutor`.

**Review by pei-jung — CHANGES_REQUESTED** [review]
@ctang3 the ticket number should be **LPD-15348**. Use LPD instead of LPS. Please update the commit message. Thanks!

File: modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/notifications/ScheduledPublicationUserNotificationHandler.java (hunk 1 above)
Comment (ctang3) [reply modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/notifications/ScheduledPublicationUserNotificationHandler.java:67]: I went through looking to escape instances of the ctCollectionName being used in notification-related strings. It seemed like this is separate from `CTPublishBackgroundTaskExecutor`. Does data from `ScheduledPublicationUserNotificationHandler` get passed through `CTPublishBackgroundTaskExecutor` at some point that I'm missing?

**Comment by ctang3** [comment]
Updated the commit message - S and D are too close together for my muscle memory 😅

**Review by pei-jung — APPROVED** [review]

**Comment by pei-jung** [comment]
ci:test:sf

**Comment by pei-jung** [comment]
@ctang3 looks like there's a merge conflict. Can you rebase again?

**Comment by ctang3** [comment]
ci:test:sf

**Comment by pei-jung** [comment]
ci:test:sf

**Comment by pei-jung** [comment]
@xbrianlee can you help me with the SF failure on this PR? It's not clear to me what the failure is about.

**Comment by xbrianlee** [comment]
@pei-jung Here's the error:
```
     [java] The commit message contains the word 'xss', which could reveal potential security vulnerablities. Please see the vulnerability keywords that are specified in source-formatter.properties in the liferay-portal repository.
     [java] 	at com.liferay.source.formatter.SourceFormatter._validateCommitMessages(SourceFormatter.java:1309)
     [java] 	at com.liferay.source.formatter.SourceFormatter.format(SourceFormatter.java:327)
     [java] 	at com.liferay.source.formatter.SourceFormatter.main(SourceFormatter.java:291)
```

[These words](https://github.com/brianchandotcom/liferay-portal/blob/12924605b34672c99a0be029467402f82de533bc/source-formatter.properties#L137-L156) are not allowed in commit messages due to security concerns.

**Comment by pei-jung** [comment]
Thanks @xbrianlee! Sorry I missed it from the console log. Usually the error will be shown in the details when I click on the `Click here for more details`.

**Comment by pei-jung** [comment]
@ctang3 can you update the commit message and PR title to fix the SF error?

**Comment by xbrianlee** [comment]
Np! This run also encountered local git server errors which is why the error snippet looks like a CI error. I got confused by that also 😅

**Comment by ctang3** [comment]
ci:test:sf

**Comment by pei-jung** [comment]
ci:forward

**Comment by pei-jung** [comment]
ci:forward:force

**Comment by pei-jung** [comment]
ci:forward:force

**Pull request closed without merging**